include/deal.II/base/mpi.h

```diff
@@ -268,6 +272,71 @@ namespace Utilities
       MPI_Comm comm;
     };
 
+    /**
+     * An implementation of a critical section that guards an algorithm with MPI
```

Comment:
I would mention it explicitly here that it works for one scope (similar as `TimerOutput::Scope`).

Reply:
The term you use here and in the function name is wrong: It's the *guard* for a critical section, not the critical section itself. (As in, for example, `std::lock_guard`.) I'd name the class `CriticalSectionGuard`. But that's still not quite specific enough, because you really only use the class for the particular case of P2P communication, so maybe `P2PCriticalSectionGuard`?